corehq/form_processor/models.py

```diff
@@ -892,7 +892,8 @@ def __unicode__(self):
 
     class Meta:
         index_together = [
-            ["domain", "owner_id", "closed"],
+            ["owner_id", "closed"],  # pghero recommended just indexing owner_id, but that's probably because there's few closed cases
+            ["owner_id", "server_modified_on"],
```

Comment:
To me this means that we're doing a lot of queries by owner id without including a domain in the query. (If there were a `WHERE domain = '%s'` clause in the query, then `["domain", "owner_id", ...]` would be the right index to use.) Not including a domain in a db query to me seem like an obvious read security vulnerability—i.e. it could easily (in combination with other easy-to-make oversights) result in a simple way for some to gain access to a case in another domain. (Not saying I have an obvious one in mind, just that it seems pretty plausible.)

Thoughts?